dev-tools/xtask/src/a4x2_deploy.rs

```diff
@@ -0,0 +1,472 @@
+    #[clap(long, default_value_t = Utf8PathBuf::from(DEFAULT_A4X2_PKG_PATH))]
+    package: Utf8PathBuf,
+}
+
+#[derive(Args, Clone)]
+pub struct RunTestsArgs {
+    /// Execute omicron live tests
+    #[clap(long)]
+    live_tests: bool,
+
+    /// Leave a4x2 running after tests. Useful for investigating test
+    /// failures.
+    #[clap(long)]
+    leave_running: bool,
+
+    /// Path to an a4x2-package bundle generated by a4x2-package xtask.
+    #[clap(long, default_value_t = Utf8PathBuf::from(DEFAULT_A4X2_PKG_PATH))]
+    package: Utf8PathBuf,
+}
+
+struct Environment {
+    a4x2_package_tar: Utf8PathBuf,
+    work_dir: Utf8PathBuf,
+    a4x2_dir: Utf8PathBuf,
+    out_dir: Utf8PathBuf,
+    in_ci: bool,
+}
+
+pub fn run_cmd(args: A4x2DeployArgs) -> Result<()> {
+    let sh = Shell::new()?;
+
+    // TODO: do we want to have a configurable falcon dataset?
+
+    let env = {
+        // Buildomat sets the CI environment variable. We'll work a bit
+        // differently in that case (no cleanup, use /work)
+        let in_ci = env::var("CI").is_ok();
+
+        let a4x2_package_tar = match &args.command {
+            // Unused by stop command, so harmless to fill in with a default
+            DeployCommand::Stop => Utf8PathBuf::from(DEFAULT_A4X2_PKG_PATH),
+
+            DeployCommand::Start(args) => args.package.clone(),
+            DeployCommand::RunTests(args) => args.package.clone(),
+        };
+        let a4x2_package_tar = a4x2_package_tar.canonicalize_utf8()?;
+
+        let home_dir = Utf8PathBuf::from(env::var("HOME")?);
+        let work_dir = if in_ci {
+            Utf8PathBuf::from("/work/a4x2-deploy")
+        } else {
+            home_dir.join(".cache/a4x2-deploy")
+        };
+
+        // a4x2 dir. will be created by the unpack step
+        let a4x2_dir = work_dir.join("a4x2-package-out");
+
+        // Output. Maybe in CI we want this to be /out
+        let out_dir = work_dir.join("a4x2-deploy-out");
+
+        Environment { a4x2_package_tar, work_dir, a4x2_dir, out_dir, in_ci }
+    };
+
+    match args.command {
+        DeployCommand::Stop => teardown_a4x2(&sh, &env)?,
+        DeployCommand::Start(_) | DeployCommand::RunTests(_) => {
+            if !env.in_ci {
+                // Teardown previous deploy if it exists, before wiping the data
+                // for it. If this errors, we assume the deploy doesn't exist
+                // and carry on.
+                let result = teardown_a4x2(&sh, &env);
+                eprintln!("teardown result: {:?}", result);
+                eprintln!("continuing regardless of whether there were errors");
+
+                // Delete any results from previous runs. We don't mind if
+                // there's errors. This needs to run as root in case there are
+                // artifacts owned by root left around from the deploy.
+                cmd!(sh, "pfexec rm -rf").arg(&env.work_dir).run()?;
```

Comment:
This is (obviously) incredibly dangerous.  I wonder how we can make it less so.

Reply:
Yeah, agreed -- if you could set `work_dir` to something like `/etc` (e.g. via the CLI option discussed above) you could nuke the whole system.

How many files is this typically, and how many are owned by root? If it's not too many, I'm wondering if we could do a first pass that iterates over the file system to find files owned by root, and then lists out those files. Then the user can be prompted to ask if it's okay to use `pfexec` (and there can be a `--force` option to skip prompting.)